tests/model_registry/model_catalog/upgrade/test_catalog_upgrade.py

```diff
@@ -0,0 +1,429 @@
+        assert custom_catalog_found, (
+            f"Custom catalog {TEST_DATA['catalog_id']} not found in sources: {[s['id'] for s in sources_response['items']]}"  # noqa: E501
+        )
+
+    @pytest.mark.pre_upgrade
+    def test_verify_default_catalog_preserved_pre_upgrade(
+        self,
+        model_catalog_rest_url: list[str],
+        model_registry_rest_headers: dict[str, str],
+    ):
+        """Verify default catalog still exists alongside custom catalog"""
+        sources_response = self._get_sources_response(
+            model_catalog_rest_url=model_catalog_rest_url, model_registry_rest_headers=model_registry_rest_headers
+        )
+
+        default_catalog_found = any(source["id"] == DEFAULT_CATALOG_ID for source in sources_response["items"])
+        assert default_catalog_found, (
+            f"Default catalog should still exist alongside custom catalog. Available sources: {[s['id'] for s in sources_response['items']]}"  # noqa: E501
+        )
+
+    @pytest.mark.pre_upgrade
+    def test_verify_custom_model_accessible_pre_upgrade(
+        self,
+        model_catalog_rest_url: list[str],
+        model_registry_rest_headers: dict[str, str],
+    ):
+        """Verify each model in custom catalog is accessible"""
+        # Test first model from TEST_DATA (keeping it simple with one assert)
+        model_name = TEST_DATA["models"][0]
+        try:
+            model_response = execute_get_command(
+                url=f"{model_catalog_rest_url[0]}sources/{TEST_DATA['catalog_id']}/models/{model_name}",
+                headers=model_registry_rest_headers,
+            )
+            assert model_response["name"] == model_name, (
+                f"Model name mismatch: expected {model_name}, got {model_response.get('name')}"
+            )
+        except ResourceNotFoundError as e:
+            pytest.fail(f"Model {model_name} not found in custom catalog {TEST_DATA['catalog_id']}: {e}")
+
+    @pytest.mark.pre_upgrade
+    def test_verify_catalog_reconciliation_integrity_pre_upgrade(
+        self,
+        admin_client: DynamicClient,
+        model_registry_namespace: str,
+    ):
+        """Verify both catalogs are present in configmap with proper integrity"""
+        current_configmap_data = ConfigMap(
+            name=DEFAULT_MODEL_CATALOG, client=admin_client, namespace=model_registry_namespace
+        ).instance.data
+
+        # Parse the current sources.yaml to verify structure
+        current_catalogs = yaml.safe_load(current_configmap_data["sources.yaml"])["catalogs"]
+        catalog_ids = [catalog["id"] for catalog in current_catalogs]
+
+        # Single assertion: verify we have at least default and custom catalogs
+        assert len(catalog_ids) >= 2 and DEFAULT_CATALOG_ID in catalog_ids and TEST_DATA["catalog_id"] in catalog_ids, (
+            f"Should have at least default and custom catalogs. Found: {catalog_ids}"
+        )
```

Comment:
This comes under pre-upgrade regression. in 2.25 we already have these tests. Upgrade tests should build on top of that assumption that regression is covered.